src/plugins/lightscript.js

```diff
@@ -65,64 +65,62 @@ pp.rewriteOperator = function (node) {
   }
 };
 
-pp.parseForFrom = function (node, iterator) {
-  this.expectContextual("from");
-  // `for i from`
-  const arrayOrRangeStart = this.parseExpression(true);
-  if (this.match(tt._thru) || this.match(tt._til)) {
-    return this.parseForFromRange(node, iterator, arrayOrRangeStart);
+function getIterableForKeyword(name) {
+  if (name === "idx" || name === "elem") {
+    return "array";
+  } else if (name === "key" || name === "val") {
+    return "object";
   } else {
-    return this.parseForFromArray(node, iterator, null, arrayOrRangeStart);
+    return null;
   }
-};
+}
 
-pp.parseForFromArray = function (node, iterator, element = null, array = null) {
-  if (iterator.type !== "Identifier") this.unexpected(iterator.start);
-  if (element && element.type !== "Identifier") this.unexpected(element.start);
-  node.id = iterator;
-  node.elem = element;
+pp.parseEnhancedForIn = function (node) {
+  const firstIterationType = this.state.value;
+  const iterableType = getIterableForKeyword(firstIterationType);
 
-  if (array) {
-    node.array = array;
-  } else {
-    this.expectContextual("from");
-    node.array = this.parseMaybeAssign(true);
-  }
+  this.next();
+  this.check(tt.name);
+  const firstIterationVar = this.state.value;
+  this.next();
```

Comment:
I think you can replace above three lines with 

```
const firstIterationVar = this.expect(tt.name);
```

it'll give you an identifier node instead of a string, which is a good thing.